redaxo/src/addons/debug/lib/sql_debug.php

```diff
@@ -1,26 +1,22 @@
 <?php
 
-rex_extension::register('OUTPUT_FILTER', ['rex_sql_debug', 'doLog']);
-
 /**
- * Class to monitor sql queries.
- *
- * @author staabm
- *
  * @package redaxo\debug
+ *
+ * @internal
  */
 class rex_sql_debug extends rex_sql
 {
-    private static $queries = [];
-    private static $errors = 0;
-
     /**
      * {@inheritdoc}.
      */
     public function setQuery($qry, array $params = [], array $options = [])
     {
         try {
+            $timer = new rex_timer();
             parent::setQuery($qry, $params, $options);
+            rex_debug::getInstance()
+                ->addDatabaseQuery($qry, $params, $timer->getDelta(), ['connection' => $this->DBID] + rex_debug::getTrace());
```

Comment:
rex_sql::setQuery landet teilweise in rex_sql::execute, deswegen werden manche Queries doppelt geloggt.
(wenn ich das richtig sehe)

Reply:
hab ne Bedingung eingebaut um zu checken, ob darin landet.